cli/src/main/java/com/devonfw/tools/ide/property/Property.java

```diff
@@ -170,51 +178,199 @@ public String getValueAsString() {
    * @param valueToFormat the value to format.
    * @return the given {@code value} formatted as {@link String}.
    */
-  protected String format(T valueToFormat) {
+  protected String format(V valueToFormat) {
 
-    return this.value.toString();
+    return valueToFormat.toString();
   }
 
   /**
    * @param value the new {@link #getValue() value} to set.
    * @see #getValue()
    */
-  public void setValue(T value) {
+  public void setValue(V value) {
 
     this.value = value;
   }
 
   /**
    * @param valueAsString the new {@link #getValue() value} as {@link String}.
+   * @param context the {@link IdeContext}
    * @see #getValueAsString()
    */
-  public void setValueAsString(String valueAsString) {
+  public void setValueAsString(String valueAsString, IdeContext context) {
 
     if (valueAsString == null) {
       this.value = getNullValue();
     } else {
-      this.value = parse(valueAsString);
+      this.value = parse(valueAsString, context);
+    }
+  }
+
+  /**
+   * Like {@link #setValueAsString(String, IdeContext)} but with exception handling.
+   *
+   * @param valueAsString the new {@link #getValue() value} as {@link String}.
+   * @param context the {@link IdeContext}
+   * @param commandlet the {@link Commandlet} owning this property.
+   * @return {@code true} if the value has been assigned successfully, {@code false} otherwise (an error occurred).
+   */
+  public final boolean assignValueAsString(String valueAsString, IdeContext context, Commandlet commandlet) {
+
+    try {
+      setValueAsString(valueAsString, context);
+      return true;
+    } catch (Exception e) {
+      String message = INVALID_ARGUMENT_WITH_CAUSE;
+      if (e instanceof IllegalArgumentException) {
+        message = INVALID_ARGUMENT;
+      }
+      context.warning(message, valueAsString, getNameOrAlias(), commandlet.getName(), e.getMessage());
+      return false;
     }
   }
 
   /**
    * @return the {@code null} value.
    */
-  protected T getNullValue() {
+  protected V getNullValue() {
 
     return null;
   }
 
   /**
    * @param valueAsString the value to parse given as {@link String}.
+   * @param context the {@link IdeContext}.
    * @return the parsed value.
    */
-  public abstract T parse(String valueAsString);
+  public abstract V parse(String valueAsString, IdeContext context);
 
-  @Override
-  public int hashCode() {
+  /**
+   * @param args the {@link CliArguments} already {@link CliArguments#current() pointing} the {@link CliArgument} to
+   *        apply.
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Commandlet} owning this property.
+   * @param collector the {@link CompletionCandidateCollector}.
+   * @return {@code true} if it matches, {@code false} otherwise.
+   */
+  public boolean apply(CliArguments args, IdeContext context, Commandlet commandlet,
```

Comment:
Maybe add a unit test for this method.

Reply:
Same as above. However, tested indirectly. As this method is implementing both regular CLI argument parsing/assignment and auto-completion it may be hard to write a good test only based on `Property` without further integrations. We could do that but since it is and will be covered by more integrated tests, I do not plan further dedicated tests on this method. We can create a ticket for it, if you still see the need.